packages/core/src/App/Containers/AccountSwitcherWallet/account-switcher-wallet-item.tsx

```diff
@@ -36,19 +35,19 @@ export const AccountSwitcherWalletItem = observer(
 
         const theme = is_dark_mode_on ? 'dark' : 'light';
         const app_icon = is_dark_mode_on ? 'IcWalletOptionsDark' : 'IcWalletOptionsLight';
-        const is_active = loginid === active_loginid;
-        const is_selected = is_active || linked_to?.some(account => account.loginid === active_loginid);
+        const is_dtrade_active = dtrade_loginid === active_loginid;
 
         const onAccountSwitch = async () => {
             closeAccountsDialog();
-            if (is_selected) return;
+            if (is_dtrade_active) return;
             await switchAccount(dtrade_loginid);
+            localStorage.setItem('active_wallet_loginid', loginid);
```

Comment:
as per other comment - seems like we do have to maintian account switching both via core and wallets, so at least worth extracting the name somewhere